### PR TITLE
Cleanup warnings Desktop + Mobile

### PR DESCRIPTION
**Pull request by MarcoPolo**
This fixes a bug in the desktop tab system that caused it to rerender the same thing across multiple tabs.

Fixes all the invalid prop warning from Navigator.

Cleans up other warnings.

Now the only warning that shows up is because of redux-devtools.

@keybase/react-hackers

### CODE REVIEW

File: react-native/react/tabs/devices/index.js [hunk 1]
```diff
@@ -47,7 +50,7 @@ export default class Devices extends Component {
   }
 
   render () {
-    return <DevicesRender devices={this.props.devices} showRemoveDevicePage={this.props.showRemoveDevicePage} showExistingDevicePage={this.props.showExistingDevicePage} showGenPaperKeyPage={this.props.showGenPaperKeyPage} />
+    return <DevicesRender {... this.props}/>
```
Comment (chrisnojima) [inline react-native/react/tabs/devices/index.js]: let's not use the ...this.props construct

**Comment by chrisnojima** [comment]
one comment, then :+1:

File: react-native/react/tabs/devices/index.js [hunk 2]
```diff
@@ -47,12 +50,27 @@ export default class Devices extends Component {
...
+      showGenPaperKeyPage
+    } = this.props
+
+    return <DevicesRender {... {
+      devices,
+      waitingForServer,
+      showRemoveDevicePage,
+      showExistingDevicePage,
+      showGenPaperKeyPage
+    }}/>
```
Comment (MarcoPolo) [inline react-native/react/tabs/devices/index.js]: @chrisnojima what do you think about this format?

Comment (chrisnojima) [inline react-native/react/tabs/devices/index.js]: i dunno, i debated doing this during lunch but it doesn't really seems more clear / clean to me than just doing
devices={this.props.devices} honestly

Comment (chrisnojima) [inline react-native/react/tabs/devices/index.js]: especially when its mixed with derived props and etc. exploding / imploding it i dunno, double the lines...

Comment (chrisnojima) [inline react-native/react/tabs/devices/index.js]: Example: https://github.com/keybase/client/pull/1328/files#diff-a2c60d76a656506609e969e7a0e61e3cR117

**Pull request merged**